diff --git a/web/sites/example.settings.local.php b/web/sites/example.settings.local.php
index 73671ab..89921e1 100644
--- a/web/sites/example.settings.local.php
+++ b/web/sites/example.settings.local.php
@@ -153,3 +153,16 @@
  * the language or field module.
  */
 # $settings['config_exclude_modules'] = ['devel', 'stage_file_proxy'];
+
+
+/** Customize to your local configuration */
+$databases['default']['default'] = array (
+  'database' => 'drupal',
+  'username' => 'root',
+  'password' => 'root',
+  'prefix' => '',
+  'host' => 'localhost',
+  'port' => '3306',
+  'namespace' => 'Drupal\\Core\\Database\\Driver\\mysql',
+  'driver' => 'mysql',
+);
